packages/server/lib/browsers/browser-cri-client.ts

```diff
@@ -573,6 +573,18 @@ export class BrowserCriClient {
     this.extraTargetClients.delete(targetId)
   }
 
+  async closeExtraTargets () {
+    for (const [targetId] of this.extraTargetClients) {
```

Comment:
I think this for loop with the awaited promise will be blocking -- would it be better to map these and use Promise.all to ensure each completes if we do need to wait on these?

Reply:
It's intended to be blocking and will await all of them as-is. Using Promise.all would run them in parallel, but I'm not sure that's advisable with CDP. I think it's a safer bet to run them serially.

Reply:
Why wouldn't that be advisable with CDP? I think using `Promise.all` makes sense here as well.

Reply:
I thought there was a case where CDP had trouble with too many commands being run at once, but I can't find that instance. I updated it to [run the target closing in parallel](https://github.com/cypress-io/cypress/pull/28204/commits/8df862a97821a7cede657f19aba9c652ceeca09a).